### PR TITLE
wip: Improve iso build job

### PR DESCRIPTION
**Pull request by dalehamel**
Building ISO image seems broken

### CODE REVIEW

**Comment by dalehamel** [comment]
It looks like the whole run doesn't need to happen on a self-hosted runner, probably just the functional test.

Perhaps the build iso workflow can be run on github, and just the tests done on the self-hosted runner?

**Comment by dalehamel** [comment]
I've run into the same error as https://github.com/kubernetes/minikube/pull/9329:

```
2020-10-06T00:05:29.7410097Z # runtime/internal/atomic
2020-10-06T00:05:29.7410851Z compile: version "go1.14.6" does not match go tool version "go1.11.13"
2020-10-06T00:05:29.7432863Z # internal/cpu
2020-10-06T00:05:29.7433371Z compile: version "go1.14.6" does not match go tool version "go1.11.13"
2020-10-06T00:05:29.7461244Z # runtime/internal/sys
2020-10-06T00:05:29.7461656Z compile: version "go1.14.6" does not match go tool version "go1.11.13"
2020-10-06T00:05:29.7494096Z # internal/race
2020-10-06T00:05:29.7494667Z compile: version "go1.14.6" does not match go tool version "go1.11.13"
2020-10-06T00:05:29.7524436Z # sync/atomic
2020-10-06T00:05:29.7525794Z compile: version "go1.14.6" does not match go tool version "go1.11.13"
2020-10-06T00:05:29.7565446Z # math/bits
2020-10-06T00:05:29.7565971Z compile: version "go1.14.6" does not match go tool version "go1.11.13"
2020-10-06T00:05:29.7566536Z # unicode/utf8
2020-10-06T00:05:29.7566873Z compile: version "go1.14.6" does not match go tool version "go1.11.13"
2020-10-06T00:05:29.7598026Z # encoding
2020-10-06T00:05:29.7598671Z compile: version "go1.14.6" does not match go tool version "go1.11.13"
2020-10-06T00:05:29.7606268Z # unicode
2020-10-06T00:05:29.7606622Z compile: version "go1.14.6" does not match go tool version "go1.11.13"
2020-10-06T00:05:29.7628747Z # unicode/utf16
2020-10-06T00:05:29.7629113Z compile: version "go1.14.6" does not match go tool version "go1.11.13"
2020-10-06T00:05:29.7665220Z # internal/nettrace
2020-10-06T00:05:29.7671054Z compile: version "go1.14.6" does not match go tool version "go1.11.13"
2020-10-06T00:05:29.7726582Z make[1]: *** [/home/runner/work/minikube/minikube/out/buildroot/output/build/cni-v0.7.1/.stamp_built] Error 2
2020-10-06T00:05:29.7727596Z package/pkg-generic.mk:252: recipe for target '/home/runner/work/minikube/minikube/out/buildroot/output/build/cni-v0.7.1/.stamp_built' failed
2020-10-06T00:05:29.7729085Z make[1]: Leaving directory '/home/runner/work/minikube/minikube/out/buildroot'
2020-10-06T00:05:29.7762377Z make: *** [minikube_iso] Error 2
2020-10-06T00:05:29.7763210Z Makefile:230: recipe for target 'minikube_iso' failed
2020-10-06T00:05:29.7768905Z ##[error]Process completed with exit code 2.
2020-10-06T00:05:29.7861616Z Post job cleanup.
2020-10-06T00:05:30.2009649Z [command]/usr/bin/git version
2020-10-06T00:05:30.2062162Z git version 2.28.0
2020-10-06T00:05:30.2104282Z [command]/usr/bin/git config --local --name-only --get-regexp core\.sshCommand
2020-10-06T00:05:30.2155938Z [command]/usr/bin/git submodule foreach --recursive git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :
2020-10-06T00:05:30.2708251Z [command]/usr/bin/git config --local --name-only --get-regexp http\.https\:\/\/github\.com\/\.extraheader
2020-10-06T00:05:30.2741174Z http.https://github.com/.extraheader
2020-10-06T00:05:30.2755205Z [command]/usr/bin/git config --local --unset-all http.https://github.com/.extraheader
2020-10-06T00:05:30.2786940Z [command]/usr/bin/git submodule foreach --recursive git config --local --name-only --get-regexp 'http\.https\:\/\/github\.com\/\.extraheader' && git config --local --unset-all 'http.https://github.com/.extraheader' || :
2020-10-06T00:05:30.3055633Z Cleaning up orphan processes

```

**Comment by afbjorklund** [comment]
It _would_ have been better to cherry-pick those build fixes into a separate PR, for sure

https://github.com/kubernetes/minikube/pull/9329/commits/46ea280cd363f228d1e922816e208fe1344b0439 iso.log

https://github.com/kubernetes/minikube/pull/9329/commits/d8e39a2dc531528266f4fa537fca70c0f206f40c go target

I was under the impression that it was going to merge soon, before the release work started...

https://github.com/kubernetes/minikube/milestone/33 `v1.14.0`

But I also don't want to revert it (again), and want to update OS.

**Comment by afbjorklund** [comment]
> Building ISO image seems broken

I fail to see what the main problem here was (besides the log not being uploaded when the build failed)

The ISO build log is a bit on the big side (50M), so better to store it as an artifact than to stream it out...

**Review by afbjorklund — CHANGES_REQUESTED** [review]
There's other ongoing changes on the same topic

**Comment by dalehamel** [comment]
@afbjorklund ignore this, it's a draft PR trying to debug why the build was failing for https://github.com/kubernetes/minikube/pull/9393, since I couldn't see the log output. I've been pushing the branch to try and get output.

Streaming to stdout is to be able to watch the build, github actions has no issue storing this so it seems better to me than a 50 minute long task with no output.

File: .github/workflows/iso.yml [hunk 1]
```diff
@@ -3,6 +3,7 @@ on:
   pull_request:
     paths:
       - "deploy/iso/**"
+      - ".github/workflows/**"
```
Comment (medyagh) [inline .github/workflows/iso.yml:6]: how about changing this to only the  .github/workflows/iso.yml 

so it doesn't get trigered when we edit the other workflows ?

Comment (dalehamel) [reply .github/workflows/iso.yml:6]: good idea

**Comment by priyawadhwa** [comment]
Hey @dalehamel are you still working on this?

**Comment by medyagh** [comment]
@dalehamel still working on this ?

**Comment by dalehamel** [comment]
@medyagh @priyawadhwa yes I intend to pick up this work, but it may not be for several days.

**Comment by dalehamel** [comment]
Actually I see that there is https://github.com/kubernetes/minikube/pull/9329 which better addresses the main issue, going to close this.

**Pull request closed without merging**